### ISSUE TITLE
Output of results to include object path

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule#1092]
Currently when rules report a failure (i.e. the object property is not in the desired state), the failure is reported as:

- Outcome = Fail
- One or more reasons, which are a localized message. Both custom and standard reasons are available.
- Optional, fields or data can be returned.
- The name of the rule and rule metadata.
- The source object.
- The source file and line are reported as determines by PSRule if the objects originate from a file or define a source using the `_PSRule` property.

The experience and accuracy of source file and line can vary. The current approach will return the line of the object started which could be split across multiple line, not the property that failed the test.

It would be helpful for several integration scenarios to be able to report the object path through the object to the failing property.

**Comment by BernieWhite** [comment]
@JohnathonMohr If a rule fails due to the property not existing in these cases what are your thoughts?

- `location` in the root of the object does not exist. Report `location` which is has no line or report the root object path ``.
- `properties.config2` does not exist. Report `properties` which was the last path element that was found or `properties.config2`.

```json
{
  "name": "test1",
  "properties": {
    "config1": "value1"
  }
}
```

**Comment by JohnathonMohr** [comment]
@BernieWhite I think it would be valuable to report the entire path, even if the final property (or properties) don't exist.  It could be helpful in some scenarios for the user to see exactly where a rule is making assertions, in case the property needs to be added, so it's easy to see where to add it.
